ocaml/parsing/jane_syntax_parsing.mli

```diff
@@ -118,117 +116,99 @@ module Embedded_name : sig
-  (** Which syntactic category is this for?  E.g., [module AST = Expression]. *)
-  module AST : AST
-
-  (** The type of novel syntactic terms for this syntactic category, across all
-      syntax features.  E.g., [Jane_syntax.Expression.t] *)
-  type t
-
-  (** A function to convert [Parsetree]'s AST to our novel extended one.  The
-      choice of feature and the piece of syntax will both be extracted from
-      the embedding by [Make_of_ast]'s [of_ast].
-
-      If the given syntax feature does not actually extend the given syntactic
-      category, returns [None]; this will be reported as an error. (For example:
-      There are no pattern comprehensions, so when building the extended pattern
-      AST, this function will return [None] if it spots an embedding that claims
-      to be from [Language_extension Comprehensions].)  *)
-  val of_ast_internal : Feature.t -> AST.ast -> t option
-end
-
-(** Build the [of_ast] function from [Of_ast_parameters]. The result
-    of this functor should be [include]d in modules implementing
-    [Jane_syntax.AST]. *)
-module Make_of_ast (Params : Of_ast_parameters) : sig
-  (** Interpret an AST term in the specified syntactic category as a term of the
-      appropriate auxiliary extended AST if possible.  Raises an error if it
-      finds a term from a disabled extension or if the embedding is
-      malformed. *)
-  val of_ast : Params.AST.ast -> Params.t option
+  val make_entire_jane_syntax
+    :  ('ast, 'ast_desc) t
+    -> loc:Location.t
+    -> string
+    -> (unit -> 'ast)
+    -> 'ast_desc
+
+  (** Build an [of_ast] function. The return value of this function should be
+      used to implement [of_ast] in modules satisfying the signature
+      [Jane_syntax.AST].  *)
+  val make_of_ast :
+    ('ast, _) t
+    (** Which syntactic category is this for?  E.g., [module AST = Expression].
+        'ast is the type of novel syntactic terms for this syntactic category,
+        across all syntax features. E.g., [Jane_syntax.Expression.t]
+    *)
+    -> of_ast_internal:(Feature.t -> 'ast -> 'a option)
+    (** A function to convert [Parsetree]'s AST to our novel extended one.  The
+        choice of feature and the piece of syntax will both be extracted from
+        the embedding by the first argument.
+
+        If the given syntax feature does not actually extend the given syntactic
+        category, returns [None]; this will be reported as an error. (For
+        example: There are no pattern comprehensions, so when building the
+        extended pattern AST, this function will return [None] if it spots an
+        embedding that claims to be from [Language_extension Comprehensions].)
+    *)
+    -> ('ast -> 'a option)
+      (** Interprets an AST term in the specified syntactic category as a term
+          of the appropriate auxiliary extended AST if possible. Raises an error
+          if it finds a term from a disabled extension or if the embedding is
+          malformed. *)
```

Comment:
The indentation here seems strange?

Reply:
It was meant to be a comment on the return value. I've moved it up to the function header.